services/vaults/holdings/save/handler.js

```diff
@@ -0,0 +1,111 @@
+/* 
+
+Holdings calculated here are for Vaults, Strategies and the Earn yTokens
+This is part of the TVL calculation defined here: https://hackmd.io/@dudesahn/BkxKfTzqw
+
+ */
+
+'use strict';
+
+require('dotenv').config();
+const dynamodb = require('../../../../utils/dynamoDb');
+const Web3 = require("web3");
+const moment = require("moment");
+const delay = require("delay");
+const _ = require("lodash");
+const vaults = require("./vaults");
+const EthDater = require("./ethereum-block-by-date.js");
+const { delayTime } = require("./config");
+
+const { getHoldings } = require('./getHoldings');
+
+const db = dynamodb.doc;
+const archiveNodeUrl = process.env.ARCHIVENODE_ENDPOINT;
+const infuraUrl = process.env.WEB3_ENDPOINT;
+const archiveNodeWeb3 = new Web3(archiveNodeUrl);
+const infuraWeb3 = new Web3(infuraUrl);
+const blocks = new EthDater(archiveNodeWeb3, delayTime);
+const axios = require('axios');
+
+
+
+const saveVault = async (data) => {
+  const params = {
+    TableName: "holdings",
+    Item: data,
+  };
+  await db
+    .put(params)
+    .promise()
+    .catch((err) => console.log("err", err));
+  console.log(`Saved ${data.name}`);
+};
+
+
+
+
+const readVault = async (vault) => {
+  const {
+    name,
+    symbol,
+    description,
+    vaultSymbol,
+    vaultContractABI: abi,
+    vaultContractAddress: address,
+    erc20address: tokenAddress,
+    price_id: price_id
+  } = vault;
+  console.log(`Reading vault ${vault.name}`);
+  if (!abi || !address) {
+    console.log(`Vault ABI not found: ${name}`);
+    return null;
+  }
+  try{
+    const holdings= await getHoldings(vault);
+    const priceFeed = await axios.get('https://api.coingecko.com/api/v3/coins/' + vault.price_id);
```

Comment:
can we get price on-chain from uniswap instead perhaps? this way we don't have to maintain the API

Reply:
OK, I need to investigate how to perform that. Any guidance?

Reply:
Can we do this on an improvement ticket? Unless it is a blocker, I could make the other improvements easily, deliver the TVL endpoint and then focus on changing the price feed.

Reply:
the "earn" products: yWBTC, yCRV and ybCRV are using getVirtualPrice = on-chain price.
Rest of tokens are using coingecko. I would merge as it is and create a new issue to move all pricing infro to on-chain using chainlink, for example. issue: https://github.com/iearn-finance/yearn-api/issues/34

Reply:
We should be good, we can just use Andre's keep3r price oracles, which already have all of the pairs we need. I updated with the relevant info on issue #34